src/shared/cancellations/utils.ts

```diff
@@ -14,3 +32,48 @@ export const getCancellationsSubscription = async (subscriptionId: string, table
 
     return cancellationsSubscriptionSchema.parse(subscription);
 };
+
+/**
+ * Checks if a given subscription is healthy by looking at whether any of heartbeatLastReceivedDateTime,
+ * lastResubscriptionTime, serviceStartDatetime, or lastCancellationsDataReceivedDateTime were in the last 90 seconds.
+ *
+ * Data producers are meant to send heartbeats at least every 30 seconds but this is not always the case so the extra
+ * checks are intended to prevent over re-subscribing
+ *
+ * @param subscription The subscription object to check
+ * @param currentTime The current time in DayJs
+ * @returns Whether the subscription is healthy or not
+ */
+export const checkCancellationsSubscriptionIsHealthy = (
```

Comment:
I wonder if its worth trying to create a generic function to checkSubscriptionIsHealthy to use in both AVL and cancellations since the logic is basically the same? 

In other function e.g. `getSubscriptionUsernameAndPassword` I've passed through a subscriptionType flag to help achieve this